src/powerhouse/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -9,57 +9,91 @@ import {
 } from 'react-aria-components';
 import { twMerge } from 'tailwind-merge';
 
-export interface DropdownMenuProps<T = string> {
```

Comment:
There are many instances of these generics used to handle the item id of the dropdown component. However, these generics are actually wrong and will lead to bugs. By using <T = string> in the first implementation, it would seem that the api will allow types other than strings to be used as keys. However, in the function implementation we actually do <T extends string> instead, which would then disallow using any other types than string (i.e. number etc.) and would defeat the purpose of the generic.

When investigating the type that the react aria component actually uses for its ids, which is the type that these ids must conform to, we see that it actually uses the built in React type React.Key, which maps to `string | number | bigint`. This makes sense, because those are the only values that can actually be used to identify items in this context. We can't allow non-primitive values as ids, because referential equality will mean that our keys can't be compared with ===.

All of that being said, we can eliminate all of these generics by simply using React.Key as the id type. This will let the api use numbers and strings as keys, while preventing usage of things that won't work like objects or arrays.

Reply:
The issue is that with just `React.key` we lose autocomplete functionality on the `onItemOptionsClick` handler. We could do `<T extends React.Key = React.Key>`.

https://github.com/powerhouse-inc/design-system/assets/5501872/fe860918-7c64-4923-98aa-a955fe06f35d

Of course it is debatable if it's worth it.

Reply:
I took a closer look at how the autocomplete works here. What's happening is we are using this type `DefaultOptionId` as the default argument for most of the functions, i.e. `<T extends string = DefaultOptionId>`

`export type DefaultOptionId = (typeof DefaultOptions)[number]['id']`

This is derived from an "as const" object

```
export const DefaultOptions = [
    {
        id: 'duplicate',
        label: 'Duplicate',
        icon: FilesIcon,
    },
    {
        id: 'new-folder',
        label: 'New Folder',
        icon: FolderIcon,
    },
    {
        id: 'rename',
        label: 'Rename',
        icon: PencilIcon,
    },
    {
        id: 'delete',
        label: 'Delete',
        icon: TrashIcon,
        className: 'text-[#EA4335]',
    },
] as const;
```

Which means that this autocomplete will only work in the case that the component uses these default options. For other cases it will revert back to string and lose the autocomplete anyway.

I reckon that since most users of the component will want to use their own options and will therefore still lose this autocomplete, its not worth having all of these generics.

What do you think?

Reply:
<img width="906" alt="Screenshot 2023-11-10 at 09 34 56" src="https://github.com/powerhouse-inc/design-system/assets/5501872/40b6efba-969c-4a4a-9f00-a2235f50c02c">
But the generic allows the options to be overridden. Anyway, it's not a blocker. I'll add a ticket for this small improvement and we can see when/if we pick it up.

Reply:
Created https://github.com/powerhouse-inc/design-system/issues/81